### DISCUSSION TITLE
how to pass object to query

### DISCUSSION THREAD

**Discussion by tasiotas** [Diizzayy/nuxt-graphql-client#331] [Q&A]
Hi,

Can somebody please help me with this case?
How can I pass object to a query, to define whole key/value of the field, not only value.

Here is my query:

```gql
query indexAuthors($after: String) {
  authors: users(
    first: 100
    after: $after
    filters: { hasProduct: true }
    order: { object }  # <- here I would like to pass object with key/val
  ) {
    edges {
      node {
        fullName
        slug
      }
    }
  }
}
```

`object` would be dynamically set at runtime. Here are possible ordering objects I have in mind

`{ date_published: "ASC" }`
or
`{ price: "DESC" }`
or
`{ name: "ASC" }`
and so on...

Somehow I find this difficult, getting syntax errors all the time.
Thank you

**Comment by likt0r** [comment]
I answer because i lost a few hours to figure this out.
Significant to know is that you can import the generated types from the .nuxt folder
```gql
query getAuthors($order_by: [authors_order_by!] = {}, $limit: Int = 10, $offset: Int = 10) {
  authors(order_by: $order_by, limit: $limit, offset: $offset) {
    id
    name
    books
  }
 
}
```

```typescript
import { Order_By, type Authors_Order_By } from '@/.nuxt/gql/default'

const orderBy = computed<Authors_Order_By>( () => {
 // Do something to calculate your order Object
return { 
   name: Order_By.Desc
}
})
const { data, pending } = await useAsyncGql({
  operation: 'indexAuthors',
  variables: {  order_by: orderBy }
})
```

I hope this helps if you wonder how to order ;)

Cheers
